amd_openvx/openvx/ago/ago_haf_cpu_generic_functions.cpp

```diff
@@ -0,0 +1,878 @@
+            {
+                vx_int16 slice[C_MAX_CONVOLUTION_DIM * C_MAX_CONVOLUTION_DIM] = { 0 };
+
+                auxReadRect(src_base, &src_addr, bordermode, src_format, x, y, conv_radius_x, conv_radius_y, slice);
+
+                // purpose of this section is to compensate extra terms caused by replicate border mode (it is the only one allowed)
+
+                if (y < CONV_DIM_HALF)
+                {
+                    INSERT_ZERO_Y(slice, y)
+                }
+                else if (y >= high_y - CONV_DIM_HALF)
+                {
+                    INSERT_VALUES_Y(slice, y)
+                }
+
+                if (x < CONV_DIM_HALF)
+                {
+                    INSERT_ZERO_X(slice, x)
+                }
+                else if (x >= high_x - CONV_DIM_HALF)
+                {
+                    INSERT_VALUES_X(slice, x)
+                }
+
+                for (i = 0; i < (vx_int32)(conv_width * conv_height); ++i)
+                    sum += conv_mat[conv_width * conv_height - 1 - i] * slice[i];
+            }
+
+            value = sum / (vx_int32)scale;
+
+            if (dst_format == VX_DF_IMAGE_U8)
+            {
+                vx_uint8 *dstp = (vx_uint8 *)vxFormatImagePatchAddress2d(dst_base, x, y, &dst_addr);
+                if (value < 0) *dstp = 0;
+                else if (value > UINT8_MAX) *dstp = UINT8_MAX;
+                else *dstp = value;
+            }
+            else if (dst_format == VX_DF_IMAGE_S16)
+            {
+                vx_int16 *dstp = (vx_int16 *)vxFormatImagePatchAddress2d(dst_base, x, y, &dst_addr);
+                if (value < INT16_MIN) *dstp = INT16_MIN;
+                else if (value > INT16_MAX) *dstp = INT16_MAX;
+                else *dstp = value;
+            }
+        }
+    }
+
+    status |= vxCommitImagePatch(src, NULL, 0, &src_addr, src_base);
+    status |= vxCommitImagePatch(dst, &rect, 0, &dst_addr, dst_base);
+
+    return status;
+}
+
+static vx_status upsampleImage(vx_context context, vx_uint32 width, vx_uint32 height, vx_image filling, vx_convolution conv, vx_image upsample, vx_border_t *border)
+{
+    vx_status status = VX_SUCCESS;
+    vx_df_image format, filling_format;
+
+    format = VX_DF_IMAGE_U8;
+    vx_image tmp = vxCreateImage(context, width, height, VX_DF_IMAGE_U8);
+    status |= vxQueryImage(filling, VX_IMAGE_FORMAT, &filling_format, sizeof(filling_format));
+
+    vx_rectangle_t tmp_rect, filling_rect;
+    vx_imagepatch_addressing_t tmp_addr = VX_IMAGEPATCH_ADDR_INIT;
+    vx_imagepatch_addressing_t filling_addr = VX_IMAGEPATCH_ADDR_INIT;
+    vx_map_id tmp_map_id, filling_map_id;
+    void *tmp_base = NULL;
+    void *filling_base = NULL;
+
+    status = vxGetValidRegionImage(tmp, &tmp_rect);
+    status |= vxMapImagePatch(tmp, &tmp_rect, 0, &tmp_map_id, &tmp_addr, (void **)&tmp_base, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+    status = vxGetValidRegionImage(filling, &filling_rect);
```

Comment:
Codacy found an issue: [Variable 'status' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6287409)